include/tvm/runtime/container.h

```diff
@@ -274,6 +275,99 @@ class ADT : public ObjectRef {
   TVM_DEFINE_OBJECT_REF_METHODS(ADT, ObjectRef, ADTObj);
 };
 
+/*! \brief An object representing string. It's POD type. */
+class StringObj : public Object {
+ public:
+  /*! \brief The length of the string object. */
+  uint32_t size;
+
+  /*! \brief The pointer to string data. */
+  char* data;
+
+  /*! \brief String object which is moved from std::string container. */
+  class FromStd;
+};
+
+/*! \brief An object representing string moved from std::string. */
+class StringObj::FromStd : public StringObj {
+ public:
+  /*! \brief Container that holds the memory. */
+  std::string data_container;
+};
+
+/*! \brief reference to string objects. */
+class String : public ObjectRef {
+ public:
+  /*!
+   * \brief Construct a new String object
+   *
+   * \param other The moved std::string object
+   */
+  explicit String(std::string&& other) {
+    auto ptr = make_object<StringObj::FromStd>();
+    ptr->data_container = std::move(other);
+    ptr->size = ptr->data_container.size();
+    ptr->data = const_cast<char*>(ptr->data_container.data());
+    data_ = std::move(ptr);
+  }
+
+  /*!
+   * \brief Construct a new String object
+   *
+   * \param other The source string object
+   */
+  explicit String(const std::string& other) {
+    auto ptr = make_object<StringObj::FromStd>();
+    ptr->data_container = other;
+    ptr->size = ptr->data_container.size();
+    ptr->data = const_cast<char*>(ptr->data_container.data());
+    data_ = std::move(ptr);
+  }
+
+  /*!
+   * \brief Return the length of the string
+   *
+   * \return size_t string length
```

Comment:
let use add common methods for std::string

- length- compare
  - operator==(std::string) !=
  - operator==(const char*) !=
- c_str()